File: DSA/Gautam_Shyamnani_590013078/Day45Question1.cpp
#include <iostream>
#include <queue>

using namespace std;

int sumQueue(queue<int> originalQueue) {
    int sum = 0;

    // Traverse through the queue without changing the original
    while (!originalQueue.empty()) {
        sum += originalQueue.front(); // Add the front element to sum
        originalQueue.pop();          // Remove the element (from copy only)
    }

    return sum; // Return the total sum
}

int main() {
    // Create a queue and add some elements
    queue<int> myQueue;
    myQueue.push(1);
    myQueue.push(2);
    myQueue.push(3);

    // Call the function and print the result
    int total = sumQueue(myQueue);
    cout << "Sum of queue elements: " << total << endl;

    return 0;
}

File: DSA/LakshaKundra_590013237/Day_45_Question_1.py
from queue import Queue

def sum_queue(q):
    total = 0
    size = q.qsize()
    for _ in range(size):
        val = q.get()
        total += val
        q.put(val)
    return total

q1 = Queue()
for i in [1, 2, 3]:
    q1.put(i)
print(sum_queue(q1))

q2 = Queue()
for i in [10, 20, 30]:
    q2.put(i)
print(sum_queue(q2))
